### PR TITLE
Integrate with sofa-tracer for support opentracing.

### PR DESCRIPTION
**Pull request by leizhiyuan**
### Motivation:

integration with sofa tracer
 
### Modification:

based on @ujjboy 's tracer integration, add stat info print.

in order to test .I add `com.alipay.sofa.rpc.tracer.sofatracer.factory.MemoryReporterImpl`, annotationed by @JustForTest

### Result:

we will have stat infos in rpc-client-stat.log and rpc-server-stat.log as below.

```
➜  tracelog tail -f rpc-client-stat.log
{"time":"2018-05-18 07:02:19.716","stat.key":{"method":"method","local.app":"client","service":"app.service:2.0"},"count":20,"total.cost.milliseconds":1,"success":"Y"}
{"time":"2018-05-18 07:02:19.717","stat.key":{"method":"method","local.app":"client","service":"app.service:1.0"},"count":10,"total.cost.milliseconds":17,"success":"Y"}

```
service,method,localapp,remoteapp,count,time,success.

### CODE REVIEW

File: test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java [hunk 1]
```diff
@@ -83,41 +78,34 @@ public static void beforeClass() throws IllegalAccessException, NoSuchFieldExcep
         Lookout.setRegistry(registry);
 
         RpcRunningState.setUnitTestMode(false);
+
+        try {
+            invoke();
+            Thread.sleep(5000);
```
Comment (ujjboy) [inline test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java]: 这个sleep是为了？

File: extension-impl/tracer-opentracing/src/main/java/com/alipay/sofa/rpc/tracer/sofatracer/log/tags/RpcSpanTags.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+    /**
+     * 路由记录
+     */
+    public static final String ROUTE_RECORD                 = "router.record";
+
+    /***
+     * 弹性数据位
+     */
+    public static final String ELASTIC_ID                   = "elastic.id";
```
Comment (ujjboy) [inline extension-impl/tracer-opentracing/src/main/java/com/alipay/sofa/rpc/tracer/sofatracer/log/tags/RpcSpanTags.java]: 我在想字段这几个要不要删掉？

File: test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java (hunk 1 above)
Comment (leizhiyuan) [reply test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java]: @ujjboy 为了等一些 callback 回来.

File: extension-impl/tracer-opentracing/src/main/java/com/alipay/sofa/rpc/tracer/sofatracer/log/tags/RpcSpanTags.java (hunk 2 above)
Comment (leizhiyuan) [reply extension-impl/tracer-opentracing/src/main/java/com/alipay/sofa/rpc/tracer/sofatracer/log/tags/RpcSpanTags.java]: 可以,我先删除,也没人用到现在.

File: test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java (hunk 1 above)
Comment (ujjboy) [reply test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java]: 嗯 那就在 callback 调完之后  用 CountdownLatch 等吧，这里可以不用等了。

File: test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java [hunk 3]
```diff
@@ -274,10 +280,14 @@ private void testThreadPoolQueueSize(Metric metric) {
...
      * @throws InterruptedException
      */
-    private void testProviderServiceStats(Metric metric) throws InterruptedException {
+    @Test
+    public void testProviderServiceStats() throws InterruptedException {
+
+        Metric metric = fetchWithName("rpc.provider.service.stats");
 
         for (Tag tag : metric.id().tags()) {
             if (tag.key() == "method") {
```
Comment (ujjboy) [inline test/test-integration/src/test/java/com/alipay/sofa/lookout/RpcLookoutTest.java]: 这里要改掉吧。

**Pull request merged**